test-data/unit/check-typeddict.test

```diff
@@ -761,6 +761,32 @@ reveal_type(d.get('y', None)) # E: Revealed type is 'Union[builtins.str, builtin
 [builtins fixtures/dict.pyi]
 [typing fixtures/typing-full.pyi]
 
+[case testTypedDictGetMethodTypeContext]
+# flags: --strict-optional
+from typing import List
+from mypy_extensions import TypedDict
+class A: pass
+D = TypedDict('D', {'x': List[int], 'y': int})
+d: D
+reveal_type(d.get('x', [])) # E: Revealed type is 'builtins.list[builtins.int]'
+d.get('x', ['x']) # E: List item 0 has incompatible type "str"
```

Comment:
Here mypy first infers type `List[int]` for `[...]` based on type context and then checks that all items have compatible types (which they don't). Below the type of `a` has been inferred previously so the type context is ignored -- but this is actually fine, since the formal argument type is a union that accepts anything. This is perhaps a little unintuitive, but changing this would be hard and it's not really specific to this PR. I'm testing both cases to catch regressions.